scripts/new_project.kts

```diff
@@ -9,23 +9,67 @@ object NewProject {
     private const val TEMPLATE_FOLDER_NAME = "CoroutineTemplate"
     private const val TEMPLATE_PACKAGE_NAME = "co.nimblehq.coroutine"
 
+    private const val APP_PATTERN = "^[A-Z\$][a-zA-Z0-9\$]*\$"
+    private const val PACKAGE_PATTERN = "^[a-z]+(\\.[a-z][a-z0-9]*)*\$"
+
     private val modules = listOf("app", "data", "domain")
 
     private var appName = ""
-    private val appNameWithoutSpace: String
-        get() = appName.replace(" ", "")
     private var packageName = ""
+
     private val projectPath: String
-        get() = rootPath + appNameWithoutSpace
+        get() = rootPath + appName
+
     private val rootPath: String
         get() = System.getProperty("user.dir").replace("scripts", "")
 
     fun generate(args: Array<String>) {
         handleArguments(args)
-        initializeNewProjectFolder()
-        cleanNewProjectFolder()
-        renamePackageNameFolders()
-        renamePackageNameWithinFiles()
+    }
+
+    private fun handleArguments(args: Array<String>) {
+        val agrumentError = "ERROR: Invalid Agrument name: Ensure define argruments => app-name={\"MyProject\"} package-name={com.sample.myproject}"
+        when (args.size) {
+            1 -> when {
+                args.first().startsWith(KEY_APP_NAME) -> showMessage("ERROR: No package has been provided")
+                args.first().startsWith(KEY_PACKAGE_NAME) -> showMessage("ERROR: No app name has been provided")
+                else -> showMessage(agrumentError)
+            }
+            2 -> args.map { arg ->
+                val (key, value) = arg.split(ARGUMENT_DELIMITER)
+                when (key) {
+                    KEY_APP_NAME -> validAppName(value)
+                    KEY_PACKAGE_NAME -> validPackageName(value)
+                    else -> showMessage(agrumentError)
+                }.also { executeNextSteps() }
+            }
+            else -> showMessage("ERROR: Require app-name and package-name to initialize the new project")
+        }
+    }
+
+    private fun validAppName(value: String) {
+        if (APP_PATTERN.toRegex().containsMatchIn(value)) {
+            appName = value
+        } else {
+            showMessage("ERROR: Invalid App Name: $value (needs to follow standard pattern {AppName})")
```

Comment:
I think we should terminate the process here with `System.exit(0)`. What do you think?

Reply:
Also, current implementation doesn't allow running the script as `app-name='My App'`. Please correct me if I am wrong, we allow `'My App'` as a valid `app-name` before right? 🙏 cc @Tuubz

Reply:
Correct, we should still allow spaces in `app-name` and surround it with quotes 🙏 

Reference: https://nimble-co.slack.com/archives/C03DFQD942W/p1652084187174279

Reply:
Ah, I remembered @kaungkhantsoe @Tuubz , Will update 🙏

Reply:
> I think we should terminate the process here with System.exit(0). What do you think?

@kaungkhantsoe I don't think we need to terminate the process because we didn't start the process yet. We will execute the next step**(e.g. initialize, clean package)** if all of the validation is valid and then start the process 🙏 cc @Tuubz @manh-t

Reply:
Hmmm, the process starts as soon as we run the script

I do agree with @kaungkhantsoe's idea, to stop the script/process as soon as there is an error.

Then we don't need this check:

``` private fun executeNextSteps() {
        if (appName.isNotEmpty() && packageName.isNotEmpty()) {
           ....
        }
    }
```

@Wadeewee Did we miss something here? 🙇

Reply:
@Wadeewee Something like this is what I had in mind 🙇 

Hopefully, I didn't miss any cases 

```
fun generate(args: Array<String>) {
    handleArguments(args)
    initializeNewProjectFolder()
    cleanNewProjectFolder()
    renamePackageNameFolders()
    renamePackageNameWithinFiles()
    renameApplicationClass()
    buildProjectAndRunTests()
}

private fun handleArguments(args: Array<String>) {
    if (args.size == 0) showErrorMessage("ERROR: No arguments")
    if (args.size > 2) showErrorMessage("ERROR: Too many arguments")
    args.forEach {
        val (key, value) = it.split(ARGUMENT_DELIMITER)
        when (key) {
            KEY_APP_NAME -> validateAppName(value)
            KEY_PACKAGE_NAME -> validatePackageName(value)
            else -> showErrorMessage("ERROR: Invalid argument")
        }
    }
}

private fun validateAppName(value: String) {
    if (value.isEmpty()) {
        showErrorMessage("ERROR: Missing App Name")
    } else if (!PATTERN_APP.toRegex().matches(value)) {
        showErrorMessage("ERROR: Invalid App Name")
    } else {
        appName = value
    }
}

private fun validatePackageName(value: String) {
    if (value.isEmpty()) {
        showErrorMessage("ERROR: Missing Package Name")
    } else if (!PATTERN_PACKAGE.toRegex().matches(value)) {
        showErrorMessage("ERROR: Invalid Package Name")
    } else {
        packageName = value
    }
}

private fun showErrorMessage(message: String, exitCode: Int = 0) {
    println("\n${message}\n")
    System.exit(exitCode)
}
```

Reply:
@Tuubz Missing this case, when putting only `1` argument e.g `kscript new_project.kts app-name='MyProject')` or `kscript new_project.kts package-name=com.example.myproject`

So I think let's keep the previous one for checking size and then validate the argument, Also by this way we no need to handle the empty cases in each validate function as well. 

What do you think? 🙏